code/modules/mob/living/simple_animal/guardian/guardian.dm

```diff
@@ -397,8 +422,300 @@
+	. = ..()
+
+/mob/living/simple_animal/hostile/guardian/charger/snapback()
+	if(!charging)
+		..()
+
+/mob/living/simple_animal/hostile/guardian/charger/throw_impact(atom/A)
+	if(!charging)
+		return ..()
+
+	else if(A)
+		if(isliving(A) && A != summoner)
+			var/mob/living/L = A
+			if(ishuman(A))
+				var/mob/living/carbon/human/H = A
+				L.visible_message("<span class='danger'>[src] slams into [L]!</span>", "<span class='userdanger'>[src] slams into you!</span>")
+				L.Weaken(2)
+				L.apply_damage(20, BRUTE)
+
+		charging = 0 */
+		
+//Protector
+/mob/living/simple_animal/hostile/guardian/protector
+	melee_damage_lower = 15
+	melee_damage_upper = 15
+	range = 15 //worse for it due to how it leashes
+	damage_coeff = list(BRUTE = 0.4, BURN = 0.4, TOX = 0.4, CLONE = 0.4, STAMINA = 0, OXY = 0.4)
+	playstyle_string = "<span class='holoparasite'>As a <b>protector</b> type you cause your summoner to leash to you instead of you leashing to them and have two modes; Combat Mode, where you do and take medium damage, and Protection Mode, where you do and take almost no damage, but move slightly slower.</span>"
+	magic_fluff_string = "<span class='holoparasite'>..And draw the Guardian, a stalwart protector that never leaves the side of its charge.</span>"
+	tech_fluff_string = "<span class='holoparasite'>Boot sequence complete. Protector modules loaded. Holoparasite swarm online.</span>"
+	var/toggle = FALSE
+
+/mob/living/simple_animal/hostile/guardian/protector/ex_act(severity)
+	if(severity == 1)
+		adjustBruteLoss(400) //if in protector mode, will do 20 damage and not actually necessarily kill the summoner
+	else
+		..()
+	if(toggle)
+		visible_message("<span class='danger'>The explosion glances off [src]'s energy shielding!</span>")
+
+/mob/living/simple_animal/hostile/guardian/protector/ToggleMode()
+	if(cooldown > world.time)
+		return 0
+	cooldown = world.time + 10
+	if(toggle)
+		overlays.Cut()
+		melee_damage_lower = initial(melee_damage_lower)
+		melee_damage_upper = initial(melee_damage_upper)
+		speed = initial(speed)
+		damage_coeff = list(BRUTE = 0.4, BURN = 0.4, TOX = 0.4, CLONE = 0.4, STAMINA = 0, OXY = 0.4)
+		src << "<span class='danger'><B>You switch to combat mode.</span></B>"
+		toggle = FALSE
+	else
+		var/image/I = new('icons/effects/effects.dmi', "at_shield1")
+		overlays += I
+		melee_damage_lower = 2
+		melee_damage_upper = 2
+		speed = 1
+		damage_coeff = list(BRUTE = 0.05, BURN = 0.05, TOX = 0.05, CLONE = 0.05, STAMINA = 0, OXY = 0.05) //damage? what's damage?
+		src << "<span class='danger'><B>You switch to protection mode.</span></B>"
+		toggle = TRUE
+
+/mob/living/simple_animal/hostile/guardian/protector/snapback() //snap to what? snap to the guardian!
+	if(summoner)
+		if(get_dist(get_turf(summoner),get_turf(src)) <= range)
+			return
+		else
+			src << "<span class='holoparasite'>You moved out of range, and were pulled back! You can only move [range] meters from [summoner.real_name]!</span>"
```

Comment:
This is still saying to the guardian that they are out of ranged and pulled back when you actually move the summoner